CondFormats/GEMObjects/src/GEMEMap.cc

```diff
@@ -0,0 +1,104 @@
+#include <iostream>
+#include "CondFormats/GEMObjects/interface/GEMEMap.h"
+#include "CondFormats/GEMObjects/interface/GEMROmap.h"
+
+#include <DataFormats/MuonDetId/interface/GEMDetId.h>
+
+GEMEMap::GEMEMap():
+  theVersion("") {}
+
+GEMEMap::GEMEMap(const std::string & version):
+  theVersion(version) {}
+
+GEMEMap::~GEMEMap() {}
+
+const std::string & GEMEMap::version() const{
+  return theVersion;
+}
+
+GEMROmap* GEMEMap::convert() const{
+  GEMROmap* romap=new GEMROmap();
+
+  std::vector<GEMEMap::GEMVFatMaptype>::const_iterator imap;
+  for (imap=this->theVFatMaptype.begin(); imap<this->theVFatMaptype.end();imap++){
+    for (unsigned int ix=0;ix<imap->strip_number.size();ix++){
+      GEMROmap::eCoord ec;
+      ec.vfatId= imap->vfatId[ix] - 0xf000;// chip ID is 12 bits
+      ec.channelId=imap->vfat_chnnel_number[ix];
+      GEMROmap::dCoord dc;
+      dc.stripId = 1 + imap->strip_number[ix]+(imap->iPhi[ix]-1)%3*128;
+      dc.gemDetId = GEMDetId(imap->z_direction[ix], 1, std::abs(imap->z_direction[ix]), imap->depth[ix], imap->sec[ix], imap->iEta[ix]); 
+      romap->add(ec,dc);
+      romap->add(dc,ec);
+    }
+  }
+  return romap;
+}
+
+GEMROmap* GEMEMap::convertDummy() const{
+  GEMROmap* romap=new GEMROmap();
+  uint16_t amcId = 1; //amc
+  uint16_t gebId = 1; 
+  romap->addAMC(amcId);
+	
+  for (int re = -1; re <= 1; re = re+2) {
+    for (int st = 1; st<=2; ++st) {
+      // for (int ch = GEMDetId::minChamberId; ch<=GEMDetId::maxChamberId; ++ch) {
+      // 	for (int ly = 1; ly<=GEMDetId::maxLayerId; ++ly) {
+      for (int ch = 1; ch<=36; ++ch) {
```

Comment:
There are a lot of magic numbers in this code. Please make them into descriptively-named constants.